fix: Clarify what has-root-span does (#1114)
honeycombio/refinery: rules_conditions.md

WARNING: Rules can have `Scope: trace` or `Scope: span`; `not-exists` used with `Scope: trace` will be true if **any** single span in the entire trace matches the negative condition.
This is almost never desired behavior.

### has-root-span

Tests if the trace as a whole has a root span.

The `Value` parameter can either be `true` or `false`.

> **NOTE:** `has-root-span` does not check if a given span is a root span.

Reviewer: nit - it looks like other places uses `WARNING` for calling out special characteristic of rule conditions. Maybe we can change `**NOTE**` to `WARNING:` here as well

> WARNING: `has-root-span` does not check if a given span is a root span.
> It only checks if a trace has a root span.

Reviewer 2: ```suggestion
NOTE: `has-root-span` does not check if a given span **is** a root span,
it checks if the containing trace **has** a root span.
```

Reviewer 2: Huh, my suggestion includes the opposite of Yingrong's -- I feel like "WARNING" is too strident and want this to be a note.
